### DISCUSSION TITLE
Could this be used for a gRPC endpoint method that's streaming data?

### DISCUSSION THREAD

**Discussion by c10b10** [vercel/swr#565] [General]
What if we would like to achieve realtime not by polling, but by connecting to a stream (using websockets), and persist emissions of that stream in the cache, that would be accessible cross routes.

Would `swr` be suited for that?

**Comment by sergiodxa** [comment]
If this is merged https://github.com/vercel/swr/pull/457 it will be possible, meanwhile you can do what that PR do yourself, subscribe to your WS and call mutate to update the cache of SWR.

**Reply by c10b10** [reply]
What stops it from being merged?

**Reply by sergiodxa** [reply]
An approved CR, you can ask for it there. Also add a +1 to the PR
